HIRS_Utils/src/main/java/hirs/data/persist/certificate/attributes/V2/ComponentIdentifierV2.java

```diff
@@ -0,0 +1,287 @@
+                    this.setComponentAddress(retrieveComponentAddress(addressesSequence));
+                    break;
+                case COMPONENT_PLATFORM_CERT:
+                    ASN1Sequence ciSequence = ASN1Sequence.getInstance(taggedObj, false);
+                    certificateIdentifier = new CertificateIdentifier(ciSequence);
+                    break;
+                case COMPONENT_PLATFORM_URI:
+                    ASN1Sequence uriSequence = ASN1Sequence.getInstance(taggedObj, false);
+                    this.componentPlatformUri = new URIReference(uriSequence);
+                    break;
+                case ATTRIBUTE_STATUS:
+                    ASN1Enumerated enumerated = ASN1Enumerated.getInstance(taggedObj, false);
+                    this.attributeStatus = AttributeStatus.values()[
+                            enumerated.getValue().intValue()];
+                    break;
+                default:
+                    throw new IllegalArgumentException("Component identifier contains "
+                            + "invalid tagged object.");
+            }
+        }
+    }
+
+    /**
+     * @return string for the type of component.
+     */
+    public ComponentClass getComponentClass() {
+        return componentClass;
+    }
+
+    /**
+     * @param componentClass the type of component to set
+     */
+    public void setComponentClass(final ComponentClass componentClass) {
+        this.componentClass = componentClass;
+    }
+
+    /**
+     * @return string for the certificate identifier
+     */
+    public CertificateIdentifier getCertificateIdentifier() {
+        return certificateIdentifier;
+    }
+
+    /**
+     * @param certificateIdentifier the certificate id to set
+     */
+    public void setCertificateIdentifier(final CertificateIdentifier certificateIdentifier) {
+        this.certificateIdentifier = certificateIdentifier;
+    }
+
+    /**
+     * @return the componentPlatformUri.
+     */
+    public URIReference getComponentPlatformUri() {
+        return componentPlatformUri;
+    }
+
+    /**
+     * @param componentPlatformUri the componentPlatformUri to set.
+     */
+    public void setComponentPlatformUri(final URIReference componentPlatformUri) {
+        this.componentPlatformUri = componentPlatformUri;
+    }
+
+    /**
+     * @return the attribute enumerated status.
+     */
+    public AttributeStatus getAttributeStatus() {
+        return attributeStatus;
+    }
+
+    /**
+     * @param attributeStatus the attributeState to set.
+     */
+    public void setAttributeStatus(final AttributeStatus attributeStatus) {
+        this.attributeStatus = attributeStatus;
+    }
+
+    /**
+     * @return true if the component has been modified.
+     */
+    public final boolean isModified() {
+        return getAttributeStatus() == AttributeStatus.MODIFIED;
+    }
+
+    /**
+     * @return true if the component has been removed.
+     */
+    public final boolean isRemoved() {
```

Comment:
Am I reading this right? If the status is NOT_SPECIFIED, ADDED or MODIFIED; this method returns true?

Reply:
This was a mistake.